stan/math/rev/fun/hypergeometric_pFq.hpp

```diff
@@ -0,0 +1,242 @@
+
+                             arena_b.adj() += vi.adj() * grad_b;
+                             z.adj() += vi.adj() * grad_z;
+                           });
+}
+
+/**
+ * Returns the generalised hypergeometric function applied to the
+ * input arguments.
+ *
+ * Specialization for hypergeometric_pFq(vector_d, vector_v, double)
+ *
+ * @tparam Ta Type of a argument
+ * @tparam Tb Type of b argument
+ * @tparam Tz Type of z argument
+ * @param[in] a a
+ * @param[in] b b
+ * @param[in] z z
+ * @return Generalised hypergeometric function
+ */
+template <typename Ta, typename Tb, typename Tz,
+          require_eigen_vector_vt<std::is_arithmetic, Ta>* = nullptr,
+          require_eigen_vector_vt<is_var, Tb>* = nullptr,
+          require_arithmetic_t<Tz>* = nullptr>
+inline var hypergeometric_pFq(const Ta& a, const Tb& b, const Tz& z) {
+  arena_t<promote_scalar_t<double, Ta>> arena_a = a;
+  arena_t<promote_scalar_t<var, Tb>> arena_b = b;
+  return make_callback_var(hypergeometric_pFq(arena_a, arena_b.val(), z),
+                           [arena_a, arena_b, z](auto& vi) mutable {
+                             vector_d grad_b(arena_b.size());
+
+                             grad_pFq_q(grad_b, arena_a, arena_b.val(), z);
+
+                             arena_b.adj() += vi.adj() * grad_b;
+                           });
+}
+
+/**
+ * Returns the generalised hypergeometric function applied to the
+ * input arguments.
+ *
+ * Specialization for hypergeometric_pFq(vector_d, vector_d, var)
+ *
+ * @tparam Ta Type of a argument
+ * @tparam Tb Type of b argument
+ * @tparam Tz Type of z argument
+ * @param[in] a a
+ * @param[in] b b
+ * @param[in] z z
+ * @return Generalised hypergeometric function
+ */
+template <typename Ta, typename Tb, typename Tz,
+          require_all_eigen_vector_vt<std::is_arithmetic, Ta, Tb>* = nullptr,
+          require_var_t<Tz>* = nullptr>
+inline var hypergeometric_pFq(const Ta& a, const Tb& b, const Tz& z) {
+  arena_t<promote_scalar_t<double, Ta>> arena_a = a;
+  arena_t<promote_scalar_t<double, Tb>> arena_b = b;
+  return make_callback_var(hypergeometric_pFq(arena_a, arena_b, z.val()),
+                           [arena_a, arena_b, z](auto& vi) mutable {
+                             double grad_z;
+
+                             grad_pFq_z(grad_z, arena_a, arena_b, z.val());
+
+                             z.adj() += vi.adj() * grad_z;
+                           });
+}
+
+/**
+ * Returns the generalised hypergeometric function applied to the
+ * input arguments.
+ *
+ * Specialization for hypergeometric_pFq(vector_v, vector_v, double)
+ *
+ * @tparam Ta Type of a argument
+ * @tparam Tb Type of b argument
+ * @tparam Tz Type of z argument
+ * @param[in] a a
+ * @param[in] b b
+ * @param[in] z z
+ * @return Generalised hypergeometric function
+ */
+template <typename Ta, typename Tb, typename Tz,
+          require_all_eigen_vector_vt<is_var, Ta, Tb>* = nullptr,
+          require_arithmetic_t<Tz>* = nullptr>
```

Comment:
I think these can support the new matrix type if you do a scheme like multiply with 

```suggestion
template <typename Ta, typename Tb, require_all_matrix_t<Ta, Tb>* = nullptr,
          require_all_st_var<Ta, Tb>* = nullptr,
          require_arithmetic_t<Tz>* = nullptr>
```

Or you can most likely put these all into one function with a signature like the below and use a scheme like `multiply` or `fma_reverse_pass` for deducing the arguments that need their adjoints propogated.

```cpp
template <typename Ta, typename Tb, typename Tz, require_all_matrix_t<Ta, Tb>* = nullptr,
          require_return_type_t<is_var, Ta, Tb>* = nullptr>
inline var hypergeometric_pFq(const Ta& a, const Tb& b, const Tz& z) {...}
```

Though I'm fine with leaving them each in individual functions as well

Reply:
Very good call, this cleaned up the code really nicely